server/api/crud/workflows.py

```diff
@@ -97,6 +98,7 @@ def schedule(
             workflow_request=workflow_request,
             labels=labels,
         )
+        runner.spec.node_selector = deepcopy(run_spec.spec.node_selector)
```

Comment:
I think we should either apply it on the runner or on the run. I don't see a reason to set it on both. probably having it on the run is better to make the function easy to export?